### ISSUE TITLE
Docker Machine support

### ISSUE THREAD

**Issue by Soliah** [codeclimate/codeclimate#86]
boot2docker has been deprecated in favour of Docker Machine. Doesn't seem to work though:

```
192-168-1-11.tpgi.com.au ❯❯❯ ~/Code
❯ eval "$(docker-machine env dev)"
192-168-1-11.tpgi.com.au ❯❯❯ ~/Code
❯ docker images
REPOSITORY                TAG                 IMAGE ID            CREATED             VIRTUAL SIZE
codeclimate/codeclimate   latest              81657c6d0722        8 days ago          93.25 MB
192-168-1-11.tpgi.com.au ❯❯❯ ~/Code
❯ codeclimate
Your Docker setup does not support the codeclimate wrapper script:

  > /var/run/docker.sock must exist as a Unix domain socket

We require a local Docker daemon that supports communication via the default
socket path.

Please use `docker run' to run the `codeclimate/codeclimate' image directly.

See https://github.com/codeclimate/codeclimate for more details.
```

**Comment by pbrisbin** [comment]
> boot2docker has been deprecated in favour of Docker Machine

Can you link to a source on this? We weren't aware of that.

I'm not familiar with Docker Machine, or know what it would take to get the wrapper script working in such a setup, but we can certainly look into it.

Does using `docker run` directly as the error message suggests work for your setup?

You can check [here](https://github.com/codeclimate/codeclimate/blob/master/codeclimate-wrapper#L46-L50) for example options. You'll probably need to adjust them since you don't have a `/var/run/docker.sock`.

**Comment by Soliah** [comment]
Docker run works fine but is a bit cumbersome.

By deprecated I mean the CLI is deprecated and Docker Machine will be taking over that responsibility. Docker Machine is still using the same boot2docker ISO though at this stage. 

https://github.com/boot2docker/boot2docker-cli readme indicates the deprecation.

**Comment by pbrisbin** [comment]
> Docker run works fine but is a bit cumbersome.

As a workaround, you could edit the installed wrapper script to execute the docker run command you need. If you'd like some help with that, please post a `docker run` invocation that works and I can suggest the edits you'd have to make to get that working. Seeing that example command might actually help us in adding real Docker Machine support too.

> https://github.com/boot2docker/boot2docker-cli readme indicates the deprecation.

FWIW: there is a bit of a contradiction here as the Docker Machine page linked from that one states:

> Note: Machine is currently in beta, so things are likely to change. We don’t recommend you use it in production yet.

In any event, thank you for the report. We do want to support as many use cases as possible and will absolutely support Docker Machine if that is the recommended way to run Docker containers on OS X (which it looks like it is, or is very close to being). I'll report back when we're able to prioritize this work and have a better idea on timeline.

**Comment by GordonDiggs** [comment]
This should be working now that we have merged #100!
